api/wikis/serializers.py

```diff
@@ -99,6 +108,41 @@ class NodeWikiSerializer(WikiSerializer):
         related_view_kwargs={'wiki_id': '<_id>'},
     )
 
+    def update(self, instance, validated_data):
+        auth = Auth(self.context['request'].user)
+        new_page_name = validated_data.get('page_name')
+        validated_data.pop('content', None)
+
+        if new_page_name:
+            try:
+                instance = instance.node.rename_node_wiki(instance.page_name, new_page_name, auth)
+            except (
+                NameInvalidError, NameMaximumLengthError,
+                PageCannotRenameError
+            ) as err:
+                raise ValidationError(err.message)
+            except PageConflictError as err:
+                raise Conflict(err.message)
+        return instance
+
+    def create(self, validated_data):
+        auth = Auth(self.context['request'].user)
+        node = validated_data.get('node')
+        name = validated_data.get('page_name')
+        content = validated_data.get('content', '')
+
+        if node.get_wiki_page(name=name):
+            raise Conflict("A wiki page with the name '{}' already exists.".format(name))
+
+        try:
+            wiki_page, _ = node.update_node_wiki(name=name, content=content, auth=auth)
```

Comment:
I should know this, but I remember there being some bleach stuff in the wiki code, do we need to do anything to the content before saving it to the db?

Reply:
@pattisdr, I can see where we sanitize before rendering the content, but if we sanitize on input I can't find it. @sloria do you know if this exists?

Reply:
Serializers sanitize input upon validation: https://github.com/CenterForOpenScience/osf.io/blob/86a567fcfb71c6ed03fd3611c2060203cad971e6/api/base/serializers.py#L1120

Reply:
That's fine, thanks for looking, just wanted to double-check.